### PR TITLE
Focus window on scrolling

### PR DESCRIPTION
**Pull request by nickodei**
#### Describe your PR, what does it fix/add?

This is the new PR for #8279. The other PR introduced a regression on merge which this implementation addresses as well. 

**In short**: You are now able to focus a window on scroll (like with mouse-movement and button-clicks). This particularly happens, after you switched your workspace and returned back.

What does it better? It now calls the `simulateMouseMovement` only once when the window differ and also when mouse_follow event is set to 1. This should avoid weard touchpad behaviour happening in firefox when you constantly call `simulateMouseMovement`.

#### Is it ready for merging, or does it need work?

- Tested it locally with mouse and touchpad on my Surface Pro 8
- Scrolling in Firefox works also on touchpad (was the regression)

### CODE REVIEW

File: src/managers/input/InputManager.cpp [hunk 1]
```diff
@@ -774,24 +775,32 @@ void CInputManager::onMouseWheel(IPointer::SAxisEvent e) {
...
+
+                    g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
+                    g_pSeatManager->sendPointerFrame();
+                }
+            }
 
-                g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
-                g_pSeatManager->sendPointerFrame();
+            const auto PCURRWINDOW = g_pCompositor->getWindowFromSurface(g_pCompositor->m_pLastFocus.lock());
+            if (*PFOLLOWMOUSE == 1 && PCURRWINDOW && PWINDOW != PCURRWINDOW) {
```
Comment (vaxerski) [inline src/managers/input/InputManager.cpp:802]: no {}

File: src/managers/input/InputManager.cpp [hunk 2]
```diff
@@ -774,24 +775,32 @@ void CInputManager::onMouseWheel(IPointer::SAxisEvent e) {
         const auto MOUSECOORDS = g_pInputManager->getMouseCoordsInternal();
         const auto PWINDOW     = g_pCompositor->vectorToWindowUnified(MOUSECOORDS, RESERVED_EXTENTS | INPUT_EXTENTS | ALLOW_FLOATING);
 
-        if (PWINDOW && PWINDOW->checkInputOnDecos(INPUT_TYPE_AXIS, MOUSECOORDS, e))
-            return;
+        if (PWINDOW) {
+            if (PWINDOW->checkInputOnDecos(INPUT_TYPE_AXIS, MOUSECOORDS, e)) {
```
Comment (vaxerski) [inline src/managers/input/InputManager.cpp:779]: no {}

**Review by vaxerski — COMMENTED** [review]
logic makes sense to me

File: src/managers/input/InputManager.cpp (hunk 1 above)
Comment (nickodei) [reply src/managers/input/InputManager.cpp]: got it

File: src/managers/input/InputManager.cpp (hunk 2 above)
Comment (nickodei) [reply src/managers/input/InputManager.cpp]: got it

File: src/managers/input/InputManager.cpp [hunk 3]
```diff
@@ -774,25 +775,31 @@ void CInputManager::onMouseWheel(IPointer::SAxisEvent e) {
...
+                        g_pCompositor->warpCursorTo({TEMPCURX, TEMPCURY}, true);
 
-                g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
-                g_pSeatManager->sendPointerFrame();
+                    g_pSeatManager->sendPointerMotion(e.timeMs, Vector2D{TEMPCURX, TEMPCURY} - BOX.pos());
+                    g_pSeatManager->sendPointerFrame();
+                }
             }
+
+            const auto PCURRWINDOW = g_pCompositor->getWindowFromSurface(g_pCompositor->m_pLastFocus.lock());
```
Comment (vaxerski) [inline src/managers/input/InputManager.cpp:800]: I'd generally advise against using m_pLastFocus. It's old, and not very accurate (it has 1001 quirks). Use last pointer focus from the seat manager's current state.

**Review by vaxerski — COMMENTED** [review]
rest lgtm

**Review by vaxerski — APPROVED** [review]
thanks!

**Comment by nickodei** [comment]
No Problem. I currently have the opportunity to contribute to open source projects so if there are some "manageable" issues or task, I would be down to look at them

**Comment by vaxerski** [comment]
there are 1.1k issues open, pick one you like :P

**Pull request merged**